Cabal/Distribution/Simple/PreProcess.hs

```diff
@@ -598,6 +598,7 @@ platformDefines lbi =
       PPC64       -> ["powerpc64"]
       Sparc       -> ["sparc"]
       Arm         -> ["arm"]
+      Arm64       -> ["arm64"]
```

Comment:
Can we normalize to `aarch64`? This seems to be the dominant name, unfortunately...

Reply:
*reads PR desccription* even on LLVM, I think?

Reply:
http://lists.llvm.org/pipermail/llvm-dev/2014-April/071903.html so there was two implementations of the backend in LLVM, sigh: `arm64` by Apple and `aarch64` by everyone else. They merged them shortly after the open sourcing, thank goodness.

Reply:
Note that the Linux kernel also uses `arm64`. It's a bike-shedding issue, no doubt, but personally I find that the name `arm64` conveys the information "I am an 64 bit arm cpu" nicely, whereas `aarch64` doesn't convey that kind of information and feels like an arbitrary choice that could have been "fooglbrmphf64" just as well.

Reply:
Ghc went with aarch64, because that’s the name that ARM gave the architecture.  Arm64 is something only apple came up with.

I’m aware of Linus rant regarding arm64/aarch64.

As such I would argue this should be aarch64 in cabal as well to be consistent with GHC.

I am against having aarch64 AND arm64 in Haskell.

/cc @Mistuke

Reply:
Oh haha I hadn't seen https://lkml.org/lkml/2012/7/15/133 . I agree with everything @angerman just said.

Ironically the Haskell variant name in GHC is still `ArchARM64` though, so if we just change the printing logic here can keep the variant name the same, we'll be *very* consistent :D.

Reply:
Yeah, I think we should stick with what the vendor called it, which is `AArch64` which is what the [execution state](https://developer.arm.com/products/architecture/a-profile) is called.
`AArch64` actually conveys a lot, it's an acronym for `Arm 64-bit Architecture`.  Whereas `Arm64` is ambiguous, does it refer to the `execution state` or `ISA`?

Reply:
@Mistuke when the new iteration of the naming scheme adds more indirection, stuff's going to be confusing no matter what. And then to make matters even more confusing "The AArch64 Execution state supports the A64 instruction set"

Reply:
@Ericson2314 But that's my point. Even with new additions to the naming scheme `AArch64` and `A64` will mean the same things. Just like `AArch32` and `T32`, `A32`. In this whole picture, `Arm64` has no meaning because it's not part of the specification.

Reply:
@Mistuke true. `Arm64` is definitely not needed.